proposals/runtime_on_invalidated.md

```diff
@@ -0,0 +1,156 @@
+# Proposal: browser.runtime.onInvalidated event>
+
+** How to Use This Template **
+
+See [Proposal Process](proposal_process.md) for the detailed process on how to
+propose new APIs and use this template.  Each section includes instructions on
+what to include.  Delete the instructions for a given section once it's filled
+out.  Remove this section once the template is filled out.
+
+**Summary**
+
+`browser.runtime.onInvalidated`
+
+This event allows extension frames and contentScripts to be notified when
+background scripts get invalidated.
+
+**Document Metadata**
+
+**Author:** carlosjeurissen
+
+**Sponsoring Browser:** Chrome
+
+**Contributors:** TBD
+
+**Created:** 2025-03-26
+
+**Related Issues:** https://github.com/w3c/webextensions/issues/138
+
+## Motivation
+
+### Objective
+
+This event allows extension frames and contentScripts to be notified when
+background scripts get invalidated.
+
+#### Use Cases
+
+Cleanup of effects in contentScripts. After-invalidated handling of frames.
```

Comment:
It would be nice to elaborate on these. I know what you mean, but it would unclear to someone else why `browser.tabs` wouldn't be available after the background went away, for example :)

Reply:
Sure. Elaborated a bit more in the proposal.

Reply:
Hmm, it looks the same still?

Reply:
@oliverdunk try a refresh

Reply:
Much better! Thanks.